src/v/wasm/wasmtime.cc

```diff
@@ -760,12 +819,96 @@ struct host_function<module_func> {
         mem->set_underlying_memory(&extern_item.of.memory);
         return nullptr;
     }
+
+    static wasm_trap_t* invoke_sync_host_fn_with_strict_stack_checking(
+      void* env,
+      wasmtime_caller_t* caller,
+      const wasmtime_val_t* args,
+      size_t nargs,
+      wasmtime_val_t* results,
+      size_t nresults) {
+        auto* allocator = static_cast<ss::sharded<stack_allocator>*>(env);
+        uint8_t dummy_stack_var = 0;
+        auto bounds = allocator->local().stack_bounds_for_address(
```

Comment:
cool. still on the reactor here?

Reply:
Reactor thread yes - reactor thread stack no. Wasmtime calls host functions on the VM's stack: https://docs.wasmtime.dev/api/wasmtime/struct.Config.html#method.async_stack_size

Inlined here:

> The amount of stack space guaranteed for host functions is async_stack_size - max_wasm_stack, so take care not to set these two values close to one another; doing so may cause host functions to overflow the stack and abort the process.